### ISSUE TITLE
Fix false positives for Less detached rulesets in no-extra-semicolons

### ISSUE THREAD

**Issue by thybzi** [stylelint/stylelint#2114]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

One more bug concerning LESS detached rulesets (you may hate them already :)

`no-extra-semicolons` doesn't accept the semicolon after this kind of variable ends.

> Which rule, if any, is this issue related to?

`no-extra-semicolons`

> What CSS is needed to reproduce this issue?

```less
@my-ruleset: {
    foo: bar;
    baz: qux;
};
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
    "rules": {
        "no-extra-semicolons": true
    }
}
```

> Which version of stylelint are you using?

e.g. `7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint test.less`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to LESS detached rulesets: http://lesscss.org/features/#detached-rulesets-feature

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.less
 4:2  ×  Unexpected extra semicolon   no-extra-semicolons
```

**Comment by evilebottnawi** [comment]
Reproduce

**Comment by evilebottnawi** [comment]
@davidtheclark @jeddy3 I see there are two solutions:
- I can handle this structure, know that it `rulesets` and allowed one semicolon
- I can ignore less and scss syntax (vote)

**Comment by thybzi** [comment]
I think this variant is better
> I can handle this structure, know that it `rulesets` and allowed one semicolon

**Comment by davidtheclark** [comment]
> know that it rulesets

I don't think we want to build into stylelint any parsing of non-standard syntax — no expectations that we know what it's doing, just that we know when to ignore it. I *think* this principle is consistent throughout our existing rules (is that right?).

The expectation, then, is that if Less users want to enforce semicolons in these non-standard places, they can create plugins to do so.

**Comment by evilebottnawi** [comment]
@davidtheclark agree, let's ignore this construction

**Comment by jeddy3** [comment]
> I think this principle is consistent throughout our existing rules (is that right?).

Yes it is.

> The expectation, then, is that if Less users want to enforce semicolons in these non-standard places, they can create plugins to do so.

Agreed.

> let's ignore this construction (in core)

Agreed.

**Comment by thybzi** [comment]
Would these ruleset constructions be ignored for all rules, or just this one?

**Comment by evilebottnawi** [comment]
@thybzi all (all at-rules have pattern `@ruleset:`).

**Comment by thybzi** [comment]
Understand you, thank you.

**Comment by jeddy3** [comment]
Closing as community custom syntax issue.
